### PR TITLE
Javascript/javascript1/week2

### PR DESCRIPTION
**Pull request by AnilMula**

### CODE REVIEW

File: javascript/javascript1/week2/candyHelper.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+let amountToSpend = Math.random() * 100;
+//let amountRemaining = amountToSpend; // initially amount remaing is amount to spend
+let spentMoney = 0;
+function addCandy(candyType, weight){
+    /* the function takes a candy name and weight of candy bought,
+    it calculates price for candy  and adds it to the boughtCandyPrices array */
+    let costOfCandy;
+    for(let i = 0; i < candyTablePrices.length; i++){
+        // iterate through candy prices table to calculate price of a candy
+        if(candyTablePrices[i][0] == candyType){
```
Comment (zuhairtaha) [inline javascript/javascript1/week2/candyHelper.js:12]: `===` to compare type and value

File: javascript/javascript1/week2/candyHelper.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            // check if we have enough money to by the candy
+            if(amountToSpend > costOfCandy){
+                // now we have enough money to buy the candy
+                boughtCandyPrices.push(candyType, costOfCandy);// push candy and cost of candy into an array
+                amountToSpend -= costOfCandy; //since candy we reserved: reduce the amount to spend
+            } 
+            else{
+                console.log("No more money to buy extra candies");
+                amountToSpend = 0; // when we do not have money to buy extra candies 
+                // make amouuntToSpend is ZERO, This is very importent to break the loop
```
Comment (zuhairtaha) [inline javascript/javascript1/week2/candyHelper.js:23]: you should not change the value to break the loop, what if you want to know how much do you still have?

File: javascript/javascript1/week2/candyHelper.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            }
+        }
+    }
+} 
+function canBuyMoreCandy(){
```
Comment (zuhairtaha) [inline javascript/javascript1/week2/candyHelper.js:28]: this function should have a parameter `candyType` and should return boolean value, then you can check for a specific type like this:
```js
if (canBuyMoreCandy('sweet')) {
  console.log('"You can buy more, so please do!"');
} else {
  console.log('Enough candy for you!');
}
```

File: javascript/javascript1/week2/getFullName.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+function getFullName(firstname, surname, useFormalName){
```
[the remark this answers is not in the document]
Comment (AnilMula) [reply javascript/javascript1/week2/getFullName.js:1]: yes you are right, and implemented

File: javascript/javascript1/week2/getFullName.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+    if(useFormalName == false){
```
[the remark this answers is not in the document]
Comment (AnilMula) [reply javascript/javascript1/week2/getFullName.js:2]: used this one

File: javascript/javascript1/week2/getFullName.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+        return firstname + " " + surname;
+    }
+    else{
```
[the remark this answers is not in the document]
Comment (AnilMula) [reply javascript/javascript1/week2/getFullName.js:5]: if we remove else, in some cases (when it is true)it is returned 2 times.

[the remark this answers is not in the document]
Comment (AnilMula) [reply javascript/javascript1/week2/getFullName.js:5]: no i am sorry, i implemented this

File: javascript/javascript1/week2/getEventWeekDay.js [hunk 7]
```diff
@@ -0,0 +1,12 @@
+function getEventWeekday(daysFromToday){
+    /* The purpose of this function is to return a weekday 
+    that is number of days from today */
+    // days is string array that holds all the week days 
+    const days = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"];
+    const d = new Date(); // d is an instance of an  Date object
+    let today = d.getDay(); // getday() method returns a number 0=sunday...6=saturday
```
[the remark this answers is not in the document]
Comment (AnilMula) [reply javascript/javascript1/week2/getEventWeekDay.js:7]: ok i understand , i am not updating any thing here

File: javascript/javascript1/week2/getFullName.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+        return  "lord " + firstname + " " + surname;
+    }
+}
+
+const fullname1 = getFullName("Anil","Mula",false);
+const fullname2 = getFullName("Kristen", "Threshow", true);
+console.log(fullname1);
+console.log(fullname2);
```
[the remark this answers is not in the document]
Comment (AnilMula) [reply javascript/javascript1/week2/getFullName.js:11]: yes i did this

File: javascript/javascript1/week2/studentManager.js [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+console.log(addStudentToClass("Johnny"));
+console.log(addStudentToClass("Hida"));
+console.log(addStudentToClass("Thomas"));
+console.log(addStudentToClass("Anil"));
+console.log(addStudentToClass("Tina"));
+console.log(addStudentToClass("Lia"));
+console.log(getNumberOfStudents());
+console.log(addStudentToClass("Queen"));
+console.log(addStudentToClass("Anil"));
+console.log(class07Students);
```
[the remark this answers is not in the document]
Comment (AnilMula) [reply javascript/javascript1/week2/studentManager.js:84]: yes i did this

File: javascript/javascript1/week2/candyHelper.js [hunk 10]
```diff
@@ -0,0 +1,44 @@
+let boughtCandyPrices = [];
+let candyTablePrices = [ ['Sweet', 0.5] ,[ 'Chocolate', 0.7], [ 'Toffee', 1.1],[ 'Chewing-gum',0.03]];
+let amountToSpend = Math.random() * 100;
+//let amountRemaining = amountToSpend; // initially amount remaing is amount to spend
+let spentMoney = 0;
```
[the remark this answers is not in the document]
Comment (AnilMula) [reply javascript/javascript1/week2/candyHelper.js:5]: i am updating these variables : boughtCandyPrices , amountToSpend, spentMoney.
and this is variable :candyTablePrices is constant. so i am changing type here.

File: javascript/javascript1/week2/candyHelper.js (hunk 1 above)
Comment (AnilMula) [reply javascript/javascript1/week2/candyHelper.js:12]: implemented this

File: javascript/javascript1/week2/candyHelper.js (hunk 2 above)
Comment (AnilMula) [reply javascript/javascript1/week2/candyHelper.js:23]: yes true, implemented this

File: javascript/javascript1/week2/candyHelper.js (hunk 3 above)
Comment (AnilMula) [reply javascript/javascript1/week2/candyHelper.js:28]: i have a question here, if we take only candyType as a parameter, without weight, how can we decide whether we can buy more candies. i think we need two parameters. i am not updating anything here

File: javascript/javascript1/week2/dressByWeather.js [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+        // for temperatures above 1 and below 10 degrees centigrades
+        return ["winter jacket","Sweaters", "Jumpers", "Turtlenecks"];
+    }
+    else if( temperature > 10 && temperature <= 15){
+        // for temperatures above 10 and below 15 degrees centigrades
+        return [ "Shirts", "Hoodies", "Leather jackets"];
+    }
+    else
+        // for temperatures above 16 degrees centigrades
+        return "regular clothes"
```
[the remark this answers is not in the document]
Comment (AnilMula) [reply javascript/javascript1/week2/dressByWeather.js:18]: yes true. changing this

File: javascript/javascript1/week2/dressByWeather.js [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+    }
+    else if(temperature > 0 && temperature <= 10){
+        // for temperatures above 1 and below 10 degrees centigrades
+        return ["winter jacket","Sweaters", "Jumpers", "Turtlenecks"];
+    }
+    else if( temperature > 10 && temperature <= 15){
+        // for temperatures above 10 and below 15 degrees centigrades
+        return [ "Shirts", "Hoodies", "Leather jackets"];
+    }
+    else
```
[the remark this answers is not in the document]
Comment (AnilMula) [reply javascript/javascript1/week2/dressByWeather.js:16]: yes true removing this

File: javascript/javascript1/week2/studentManager.js [hunk 13]
```diff
@@ -0,0 +1,84 @@
+let class07Students = []; // it is an empty array for a class that holds max 6 students
```
[the remark this answers is not in the document]
Comment (AnilMula) [reply javascript/javascript1/week2/studentManager.js:1]: class07Students we are using

File: javascript/javascript1/week2/studentManager.js [hunk 14]
```diff
@@ -0,0 +1,84 @@
...
+        }
+        // check if the class is full
+        if(class07Students.length == 6){
+            // now NO space for a new student
+            return "No space for a new student"; 
+        }
+        else{
+        // here class is not full just call add student()
+             addStudent(studentName);
+        }
```
[the remark this answers is not in the document]
Comment (AnilMula) [reply javascript/javascript1/week2/studentManager.js:66]: Removed inner functions, 
I used this for 2 reasons: 
1. is get used to inner functions
2. we are testing function in the code pen, so i want all in a single function.

and lastly
**if (studentName === 'queen') {
      // remove everyone and add her only
    }**

from the assignment:
"We are very fond of our Queen in Denmark, so if the Queen is added to the class she should always get a space. Even if the class has been filled out."

it doesn't say " remove everyone and add her only"

**Pull request merged**